src/main/java/Deadline.java

```diff
@@ -0,0 +1,63 @@
+import java.util.Date;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.ChronoUnit;
+
+/**
+ * Represents a deadline in the ChatBot task list.
+ * Contains date and time of due deadline and description of the task.
+ */
+public class Deadline extends Task {
+
+    public String dateTime;
+    public LocalDateTime date;
+
+    /**
+     * Constructor for the Deadline task.
+     * @param taskName description of the deadline.
+     * @param dateTime date and time when the task is due.
+     */
+    public Deadline(String taskName, String dateTime) {
+        super(taskName, "D");
+        this.dateTime = dateTime;
+        this.constructDate(dateTime);
+    }
+
+    /**
+     * Helper function to create the LocalDateTime object from the given String.
+     * @param dateTime the String that represents the date and time in the format dd-mm-yyy HH:mm.
+     */
+    public void constructDate(String dateTime) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyy HH:mm");
+        this.date = LocalDateTime.parse(this.dateTime, formatter);
+    }
+
+    @Override
+    /**
+     * The String representation of the deadline.
+     * @return String representation of the task (type, whether is it done or not, description
+     * and lastly date of due date.
+     */
+    public String toString() {
+        String message = "";
+        String dateToBePrinted = this.date.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"));
+        if (this.getDone()) {
+            message += "[" + this.getTaskType() + "]" + "[" + "✓" + "] "
+                    + this.getTaskName() + " (by: " + dateToBePrinted + ")";
+        } else {
+            message +=  "[" + this.getTaskType() + "]" + "[" + "✗" + "] "
+                    + this.getTaskName() + " (by: " + dateToBePrinted + ")";
+        }
+        return message;
+    }
+
+    /**
+     * Function to test whether the String date is in the correct format.
```

Comment:
In method header comments, the first sentence should start in the form Returns ..., Sends ..., Adds ... (not Return or Returnning etc.). So instead of starting with the word "Function", it's better to start with "Tests whether...". There should also be an empty line between description and parameter section.